S/ssht/build_tarballs.jl

```diff
@@ -3,29 +3,41 @@
 using BinaryBuilder, Pkg
 
 name = "ssht"
-version = v"1.5.2"
+ssht_version = v"1.5.2"
+# We bumped the version because we rebuilt for new architectures
+version = v"1.5.3"
 
 # Collection of sources required to complete build
 sources = [
     GitSource("https://github.com/astro-informatics/ssht.git", "c83fb2a07e1869e875e819523d2e19d91bc8a4bf"),
+    DirectorySource("bundled"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd ${WORKSPACE}/srcdir/ssht
 
+# Do not expect `-lm` to work on all systems
+atomic_patch -p1 ${WORKSPACE}/srcdir/patches/cmake_math_library.patch
```

Comment:
Is this similar to the problem you mentioned in https://github.com/JuliaPackaging/Yggdrasil/pull/10922#discussion_r2029537643?

Reply:
Not sure – the symptoms were different. But on second thought the reason could be the same. Here I thought that the system doesn't have a `libm` and that's why the build failed. I might be, again, that cmake simply didn't find the `libm`.